src/peep/peep.c

```diff
@@ -7772,13 +7772,385 @@ static void peep_on_enter_or_exit_ride(rct_peep *peep, int rideIndex, int flags)
  *
  *  rct2: 0x0069AF1E
  */
 static bool sub_69AF1E(rct_peep *peep, int rideIndex, int shopItem, money32 price)
 {
-	return !(RCT2_CALLPROC_X(0x0069AF1E, shopItem | (rideIndex << 8), 0, price, 0, (int)peep, 0, 0) & 0x100);
+#define STANDARD_FOOD_MASK 0x036BA3E0
+#define EXTRA_FOOD_MASK 0x0011FF78
+
+	rct_ride* ride = GET_RIDE(rideIndex);
+	sint16 value;
+
+	///sub_69AF1E
+	bool has_voucher;
+	RCT2_GLOBAL(0x00F1AEE3, uint8) = 0; // Seems to be a local bool for whether the peep has a matching voucher for this shop
+
+	if ((peep->item_standard_flags & PEEP_ITEM_VOUCHER) && // If they have a voucher...
+		(peep->voucher_type == VOUCHER_TYPE_FOOD_OR_DRINK_FREE) && // ...and it's a free food voucher...
+		(peep->voucher_arguments == shopItem)) { // ...and it matches the item in question?...
+		RCT2_GLOBAL(0x00F1AEE3, uint8) = 1; // then this becomes true
+		has_voucher = true;
+	}
+
+///loc_69AF47:
+	// Check to see if the peep already has the item that they're considering buying
+	int shopItem2 = shopItem; // A separate item that's used as output
+	// The root if block you see right below is common in this function.
+	// It seems to be used to branch based on whether this shop serves either a
+	// standard or an extra item. It can probably be replaced with a bool and shopItem
+	// used directly.
+	if (shopItem >= 32) {
+		shopItem2 -= 32;
+		if (peep->item_extra_flags & (1 << shopItem2)) {
+			peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_ALREADY_GOT, shopItem);
+			return 0;
+		}
+	}
+	else {
+		if (peep->item_standard_flags & (1 << shopItem2)) {
+			peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_ALREADY_GOT, shopItem);
+			return 0;
+		}
+	}
+
+///loc_69AF6B:
+	// Checks to see if the peep is already eating or is too nauseous
+	// TODO: check to make sure all of the conditions are covered
+	if (shop_item_is_food_or_drink(shopItem)) {
+		if (peep_has_food(peep)) {
+			int food = bitscanforward(peep->item_standard_flags & STANDARD_FOOD_MASK);
+			if (food != -1)
+				peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_HAVENT_FINISHED, food);
+			else
+				peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_HAVENT_FINISHED, (bitscanforward(peep->item_extra_flags & EXTRA_FOOD_MASK) + 32));
+			return 0;
+		}
+		else {
+			if (peep->nausea >= 145)
+				return 0;
+		}
+	}
+
+///loc_69AFD3:
+	// Don't buy certain items if it's raining
+	if ((shopItem == SHOP_ITEM_BALLOON) || (shopItem == SHOP_ITEM_ICE_CREAM)
+		|| (shopItem == SHOP_ITEM_COTTON_CANDY) || (shopItem == SHOP_ITEM_SUNGLASSES)) {
+		if (RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_RAIN_LEVEL, sint8) != 0)
+			return 0;
+	}
+
+///loc_69B007:
+	// Don't buy certain items if it's too cold out
+	if ((shopItem == SHOP_ITEM_SUNGLASSES) || (shopItem == SHOP_ITEM_ICE_CREAM)) {
+		if (RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_TEMPERATURE, sint8) < 12)
+			return 0;
+	}
+
+///loc_69B03B:
+	// Possible original bug: chocolate is not considered a food item for this check.
```

Comment:
I think its saying hunger does not effect if you want chocolate. I quite like the idea behind that can we add it back in. Just make it an extra if before this function.